### DISCUSSION TITLE
API Github - list contributors error on bigger contributors list

### DISCUSSION THREAD

**Discussion by joycebrum** [community/community#45433] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I am using the contributors list api to get the list of contributors from the following projects:

`GET /repos/:owner/:repo/contributors`

[github.com/torvalds/linux](https://github.com/torvalds/linux) - Requisition: `https://api.github.com/repos/torvalds/linux/contributors`
[github.com/chromium/chromium](https://github.com/chromium/chromium) - Requisition: `https://api.github.com/repos/chromium/chromium/contributors`

And it seems that internally the github api is not being able to handle the big contributors list and returns the following error:

```
Status: 403 Forbidden
{
    "message": "The history or contributor list is too large to list contributors for this repository via the API.",
    "documentation_url": "https://docs.github.com/rest/reference/repos#list-repository-contributors"
}
```

Even setting per_page as 1, the requisitions keep returning the error above.

Can't it list at least the first 500 contributors as said in [the documentation](https://docs.github.com/pt/rest/repos/repos?apiVersion=2022-11-28#list-repository-contributors)? I considered it a bug but I can actually be doing something wrong, let me know if that's the case.

**Comment by jge162** [comment]
it is expected that the API will not be able to handle the large contributors list, i don’t think that is a bug.

it's likely that the repositories you are trying to query (torvalds/linux and chromium/chromium) are indeed too large to list all contributors via the API, and the API is returning the error message as expected.

**Reply by joycebrum** [reply]
Yeah, I got that showing the full list of contributors would not be possible, but I was expecting to be able to at least see some of them. The last X contributors could be discarded and not handled by the api, this way it would still be used to get the top 10 contributors, for example.

**Comment by AndyLuo1029** [comment]
I have a similar question about this API. I want to know the exact limit of contributor count that will trigger 'too large to return' exception. I can't find this limit in documentation or exception class.
